sw/simulator/nps/nps_fdm_gazebo.cpp

```diff
@@ -83,6 +85,35 @@ static struct gazebocam_t gazebo_cams[VIDEO_THREAD_MAX_CAMERAS] =
 { { NULL, 0 } };
 #endif
 
+#ifdef NPS_SIMULATE_RANGE_SENSORS
+#include "subsystems/abi.h"
+static void gazebo_init_range_sensors(void);
+static void gazebo_read_range_sensors(void);
+#ifndef NPS_GAZEBO_RANGE_MAX_SENSORS
+#define NPS_GAZEBO_RANGE_MAX_SENSORS 255
+#endif
```

Comment:
This seems unnecessary as the number of range sensors is defined in the model and is fixed.

Reply:
That is true, but the number of sensors can not be read out until gazebo is initialized.  So this value is just to be save that it will not overflow (this amount is not the same as the number of sensors that will be selected to read out the value).

Reply:
Now you have the variable defined as REMOTE_GROUND_AMOUNT_SENSORS, this is available when you compile papaarazzi in generated/airframes.h. You can just use that